router/core/header_rule_engine.go

```diff
@@ -283,6 +287,10 @@ func (h *HeaderPropagation) OnOriginResponse(resp *http.Response, ctx RequestCon
 func (h *HeaderPropagation) applyResponseRule(propagation *responseHeaderPropagation, res *http.Response, rule *config.ResponseHeaderRule) {
 	if rule.Operation == config.HeaderRuleOperationSet {
 		propagation.header.Set(rule.Name, rule.Value)
+		if rule.Name == cacheControlKey {
+			// Handle the case where the cache control header is set explicitly
+			propagation.setCacheControl = true
```

Comment:
The refactor exposed cases where ordering of rules could adjust whether the `set` operation would successfully override CacheControl. To resolve this, I introduced this clause